### DISCUSSION TITLE
import with properties

### DISCUSSION THREAD

**Discussion by ErikNoort** [minvws/nl-kat-coordination#85] [Ideas]
Hi Openkat project,

I would like to have the option to perform imports with properties.
For example to import the clearance level with an object, the network to put an IP address in etc.

If I am overlooking the feature, please point me into the right direction.

Thanks in advance!

Best regards, Erik

**Comment by underdarknl** [comment]
Erik
We have had this question from other parties too. Importing and immediately setting clearance levels was not on our radar as we rather have OpenKAT propagate any clearance levels from related objects (like a netblock). However since we do udnerstand the usecase for this, we are working on two routes to make this possible:
First a more useful user interface to apply mass-actions on selected objects (like setting clearance levels), combined with a search for object that lack a clearance level.
Second a system where a boefje (and imports using CSV will also go trough this route in the future), are able to immediately set the clearance levels for the imported items.

The tradeoff is that any declared clearance level (on for example an ipaddress) will remain declared, even if you would undeclare the level on for example the encompassing netblock whereas they would normaly also get removed since they had inherited from that netblock.
Release 1.4 and 1.5 will bring these options online.

**Comment by underdarknl** [comment]
@ErikNoort 
The following PR should allow you to set the network for CSV added objects.
https://github.com/minvws/nl-kat-rocky/pull/95
